tests/testSphPoint.py

```diff
@@ -0,0 +1,615 @@
+
+        pointCopy = copy.deepcopy(point1)
+        self.assertEqual(pointCopy, point1)
+        self.assertEqual(point1, pointCopy)
+        self.assertFalse(pointCopy != point1)
+        self.assertFalse(point1 != pointCopy)
+
+        # Test for transitivity (may be assumed by algorithms).
+        for delta in [10.0**(0.1*x) for x in range(-150, -49, 5)]:
+            self._checkTransitive(delta)
+
+    def _checkTransitive(self, delta):
+        """Test if equality is transitive even for close points.
+
+        This test prevents misuse of approximate floating-point
+        equality -- if `__eq__` is implemented using AFP, then this
+        test will fail for some value of `delta`. Testing multiple
+        values is recommended.
+
+        Parameters
+        ----------
+        delta : `number`
+            The separation, in degrees, at which point equality may
+            become intransitive.
+        """
+        point1 = SphPoint((45.0 - delta)*degrees, -18.0*degrees);
+        point2 = SphPoint( 45.0         *degrees, -18.0*degrees);
+        point3 = SphPoint((45.0 + delta)*degrees, -18.0*degrees);
+
+        self.assertTrue(point1 != point2 or point2 != point3 or point1 == point3)
+        self.assertTrue(point3 != point1 or point1 != point2 or point3 == point2)
+        self.assertTrue(point2 == point3 or point3 != point1 or point2 == point1)
+
+    def testEqualityAlias(self):
+        """Test if == handles coordinate degeneracies correctly.
+        """
+        # Coordinate wrapping
+        self.assertEqual(SphPoint(360.0*degrees, -42.0*degrees),
+                         SphPoint(  0.0*degrees, -42.0*degrees))
+        self.assertEqual(SphPoint(-90.0*degrees, -42.0*degrees),
+                         SphPoint(270.0*degrees, -42.0*degrees))
+
+        # Polar degeneracy
+        self.assertEqual(SphPoint( 42.0*degrees,  90.0*degrees),
+                         SphPoint(270.0*degrees,  90.0*degrees))
+        self.assertEqual(SphPoint(-42.0*degrees, -90.0*degrees),
+                         SphPoint( 83.0*degrees, -90.0*degrees))
+
+        self.assertNotEqual(SphPoint(-42.0*degrees,  90.0*degrees),
+                            SphPoint( 83.0*degrees, -90.0*degrees))
+
+    def testInquality(self):
+        """Test if == and != give mutually consistent results.
+        """
+        for lon1, lat1 in self._dataset[0:20]:
+            point1 = SphPoint(lon1*degrees, lat1*degrees)
+            for lon2, lat2 in self._dataset[0:20]:
+                point2 = SphPoint(lon2*degrees, lat2*degrees)
+                self.assertTrue((point1 == point2) == (point2 == point1))
+                self.assertTrue((point1 != point2) == (point2 != point1))
+                self.assertTrue((point1 == point2) != (point1 != point2))
+
+    def testBearingToError(self):
+        """Test if bearingTo() correctly handles invalid input.
+        """
+        destination = SphPoint(42.0*degrees, -90.0*degrees)
```

Comment:
Why is `destination` at a pole? I would expect you to assert that that "north pole to safe" and south pole to safe" raise exceptions. It seems less interesting to test "north pole to destination at a pole".

Please consider eliminating `destination` and just having `northPole`, `southPole` and `safe` for simplicity and clarity. Or if you want to keep all of them, then please explain what you are doing in a code comment.